src/Interpreters/Set.cpp

```diff
@@ -278,6 +280,21 @@ void Set::checkIsCreated() const
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Trying to use set before it has been built.");
 }
 
+ColumnPtr returnFilteredColumn(const ColumnPtr & first, const ColumnPtr & second)
```

Comment:
Let's rename them to `column` and `filter`. Because `column` can have any type, but filter is only UInt8 (or `Nullable(UInt8)` or sparce)